Fill.py

```diff
@@ -129,18 +142,35 @@ def distribute_items_restrictive(world: MultiWorld, fill_locations=None):
         else:
             restitempool.append(item)
 
-    world.random.shuffle(fill_locations)
-    call_all(world, "fill_hook", progitempool, nonexcludeditempool, localrestitempool, nonlocalrestitempool, restitempool, fill_locations)
+    call_all(world, "fill_hook", progitempool, nonexcludeditempool,
+             localrestitempool, nonlocalrestitempool, restitempool, fill_locations)
+
+    locationDeficit = len(progitempool) - len(prioritizedlocations)
+    if locationDeficit > 0:
+        if locationDeficit > len(defaultlocations):
+            raise FillError(
+                f'Not enough locations for progress items. There are {len(progitempool)} progression items with {len(prioritizedlocations)} priority locations and {len(defaultlocations)} default locations')
```

Comment:
everywhere else in code they're referenced as "advancement items" not "progress items", I suggest sticking to the existing naming to reduce confusion

Reply:
Updated to match naming convention